src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -425,10 +422,19 @@ private async Task RefreshIndividualKeyValues(CancellationToken cancellationToke
                 if (_watchedSettings.TryGetValue(watchedKeyLabel, out ConfigurationSetting watchedKv))
                 {
                     KeyValueChange keyValueChange = default;
-                    await TracingUtils.CallWithRequestTracing(_requestTracingEnabled, RequestType.Watch, _requestTracingOptions,
-                        async () => keyValueChange = await _client.GetKeyValueChange(watchedKv, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                    bool success = false;
+
+                    try

```

Comment:
I'm wondering if it makes sense to wrap the whole `if...else..` (line 419- 502) in a big `try...finally`. You will call `UpdateCacheExpirationTime` just once in the end.

Reply:
I see a bigger value in minimizing the amount of code within the `try` statement. The more statements, the more time it takes to discern where the possible error comes from.

Reply:
True, if we handle errors. But the goal here is to extend cache no matter what. The code is cleaner if we use one big try....finally.

Reply:
Okay, that makes sense. Since we are not handling any exceptions, I like the idea of a single try/finally covering both failure points.